src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -0,0 +1,139 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    // Interface introduced to simplify testing.
+    internal interface ISinkManagerRegister
+    {
+        void AddSinkManager(ISinkManager manager);
+        void RemoveSinkManager(ISinkManager manager);
+    }
+
+    [Export(typeof(ISonarErrorListDataSource))]
+    internal class SonarErrorListDataSource : ITableDataSource, ISonarErrorListDataSource, ISinkManagerRegister
+    {
+        private readonly ISet<ISinkManager> managers = new HashSet<ISinkManager>();
+        private readonly ISet<ITableEntriesSnapshotFactory> factories = new HashSet<ITableEntriesSnapshotFactory>();
```

Comment:
Minor changes in lines 39 and 40 to the code extracted from the `TaggerProvider`: the sets use interfaces rather than classes.